resources/home/dnanexus/utils/generate.py

```
@@ -0,0 +1,280 @@
+        }
+
+        for future in concurrent.futures.as_completed(concurrent_jobs):
+            # access returned output as each is returned in any order
+            try:
+                data = future.result()
+                all_sample_outputs[data["sample"]] = data
+            except Exception as exc:
+                # catch any errors that might get raised during querying
+                raise RuntimeError(
+                    f"Error getting data for {concurrent_jobs[future]}: {exc}"
+                ) from exc
```

Ruff (0.8.0)

Avoid specifying long messages outside the exception class

(TRY003)